src/pygaps/characterisation/whittaker.py

```diff
@@ -0,0 +1,155 @@
+                       sorptive)
+    lambda_p = lambda_V - lambda_L
+
+    return lambda_p
+
+
+def whittaker(
+    isotherm : "ModelIsotherm",
+    loading : list = None,
+):
+    r"""
+
+    Calculate the isosteric heat of adsorption using a single isotherm via the
+    Whittaker method.
+
+    Parameters
+    ----------
+    isotherm : ModelIsotherm
+        The model isotherm used. Must be either Toth or Langmuir
+    p_sat : float
+        The saturation pressure of the sorptive, either 'real' or derived from
+        pseudo-saturation
+    loading : list[float]
+        The loadings for which to calculate the isosteric heat of adsorption
+
+    Returns
+    -------
+    df : DataFrame
+        DataFrame of isosteric heat of adsorption at input loadings.
+
+    Raises
+    ------
+    ParameterError
+        When incorrect type of model isotherm is used.
+
+    Notes
+    -----
+
+    The Whittaker method, sometimes known as the Toth potential uses variables
+    derived from fitting of a model isotherm (Langmuir or Toth) to derive the
+    isosteric enthalpy of adsorption :math:`q_{st}`. The general form of the
+    equation is;
+
+    .. math::
+        q_{st} = \Delta \lambda + \lambda_{p} + RT
+
+    Where :math:`\Delta \lambda` is the adsorption potential, and
+    :math:`\lambda_{p} is the latent heat of the liquid-vapour change at
+    equilibrium pressure.
+
+    Whittaker determined :math:`\Delta \lambda` as;
+
+    .. math::
+        \Delta \lambda = RT\ln{\left[\left(\frac{p^{sat}}{b^{\frac{1}{t}}\right)\left(\frac{\Theta^{t}}{1-\Theta^{t}}\right) \right]}
+
+    Where :math:`p^{sat}` is the saturation pressure, :math:`\Theta` is the
+    fractional coverage, and :math:`b` is derived from the equilbrium constant,
+    :math:`K` as :math:`b = \frac{1}{K^t}`. In the case that the adsorptive is
+    above is supercrictial, the pseudo saturation pressure is used;
+    :math:`p^{sat} = p_c \left(\frac{T}{T_c}\right)^2`. 
+
+    The exponent :math:`t` is only relevant to the Toth version of the method,
+    as for the Langmuir model it reduces to 1. Thus, :math:`\Delta \lambda`
+    becomes
+
+    .. math::
+        \Delta \lambda = RT\ln{\left(\frac{p^{sat}}{b^{\frac{1}{t}}\right)
+    """
+
+    if isotherm.model.name not in ['Langmuir', 'Toth']:
+        raise ParameterError('''Whittaker method requires either a Langmuir or Toth
+                         model isotherm''')
+    else:
+        R = scipy.constants.R
+        n_m = isotherm.model.params['n_m']
+        K = isotherm.model.params['K']
+        T = isotherm.temperature
+        try:
+            p_sat = isotherm.adsorbate.saturation_pressure(temp=298, unit='K')
+        except CalculationError:
+            print(f"{isotherm.adsorbate} does not have saturation pressure "
+                  f"at {isotherm.temperature} K. Calculating psduedo-saturation "
+                  f"pressure...")
+            p_c = isotherm.adsorbate.get_prop('p_critical')
```

Comment:
For some reason they don't seem to be usable? 
 
```AttributeError: Adsorbate object has no attribute 'p_critical'```

This is after my local repo has been updated to your most recent changes to adsorbates.json and adsorbate_props.json

Reply:
I keep doing so with;

```
git fetch upstream
git merge upstream/master master
```
My local fork is up to date with your master branch.

And while both `adsorbates.py`, `adsorbates.json` and `adsorbate_props.json` have been updated to include p_critical etc. they are not callable as functions. I _can_ however access this with `adsorbate.get_prop('p_critical')`. Other functions in the adsorbate class, e.g. `adsorbate.molar_mass()` are callable.

This is my .git/config
```
[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	ignorecase = true
[remote "origin"]
	url = git@github.com:sblanky/pygaps.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[branch "whittaker"]
	remote = origin
	merge = refs/heads/whittaker
[remote "upstream"]
	url = https://github.com/pauliacomi/pyGAPS
	fetch = +refs/heads/*:refs/remotes/upstream/*
```

Reply:
Well, somewhat odd, though I am at a loss as well since am not used to work with git PRs/split branches...

First, make sure you target `develop`, as this is the branch used for any ongoing work. `master` is for the last release + hotfixes 

Then, where I have actually submitted my changes is this PR, rather than the develop branch, so technically to **your** code. Looking directly at your branch, they seem to be there correctly.

Do you have your own pygaps version installed in editable mode with `pip install -e .`? Maybe the environment still has the public version.

Reply:
Yeah, sorry it was that i didn't have an editable installation. Works now!